en/lessons/basics/modules.md

```diff
@@ -1,316 +1,195 @@
----
-version: 1.4.1
-title: Modules
----
+--- version: 1.4.1 title: Modules ---
 
-We know from experience it's unruly to have all of our functions in the same file and scope.
-In this lesson we're going to cover how to group functions and define a specialized map known as a struct in order to organize our code more efficiently.
+We know from experience it's unruly to have all of our functions in the same
+file and scope. In this lesson we're going to cover how to group entities in
+modules, and group modules in components. At their most basic, modules are just
+a namespace that allow us to aggregate common pieces of functionality and avoid
+naming collisions. However, modules are use to build up more complex constructs
+called components. Modules also offer a tool for encapsulation, by allowing us
+to choose what we export.
 
 {% include toc.html %}
 
-## Modules
+## Entities, Modules, and Components
 
-Modules allow us to organize functions into a namespace.
-In addition to grouping functions, they allow us to define named and private functions which we covered in the [functions lesson](../functions/).
+In order to understand modules we have to understand the elements that compose
+a module, and what modules can be composed into. The elements that can be
+exported from a module are called entities. In Haskell those are:
 
-Let's look at a basic example:
+- Types and type synonyms
+- Values: constants and functions
+- Typeclasses
 
-```elixir
-defmodule Example do
-  def greeting(name) do
-    "Hello #{name}."
-  end
-end
+Modules are really just collections of these entities that are exported.
+Modules can be combined through a graph of imports and exports. These
+dependency graphs can be packaged together as a component. The two most common
+components are libraries and executables. A library allows you to export your
+modules for re-use in other codebases, and an executable consists of an
+entrypoint (a Main module) to the dependency graph that can be run as a
+program. A good example of a library is `base` and `prelude`. These libraries
```

Comment:
It's not clear here that `prelude` is a module exported by `base`. Changing this also likely requires changing the plurality of some other sentences.